specs/csv-to-xml.spec.js

```diff
@@ -19,42 +19,193 @@ describe('CSV to XML', function () {
         });
     });
 
-    describe('When the user converts a basic CSV file into a XML file', function () {
+    describe('Using --files flag', () => {
+        describe('When the user converts a basic CSV file into a XML file', () => {
+            beforeEach(() => {
+                return runCommand('node app.js --output=xml --files specs/data/csv-to-xml/basic-csv.csv');
+            });
+
+            afterEach(() => {
+                return deleteFileIfExists('./specs/data/csv-to-xml/basic-csv.xml');
+            });
+
+            it('Should output the XML file as expected', () => {
+                return getXml('./specs/data/csv-to-xml/basic-csv.xml').then(xml => {
+                    expect(xml.data['basic-csv'].count()).to.equal(2);
+                    expect(xml.data['basic-csv'].at(0).Col1.text()).to.equal('Hi')
+                    expect(xml.data['basic-csv'].at(0).Col2.text()).to.equal('Bye')
+                    expect(xml.data['basic-csv'].at(1).Col1.text()).to.equal('Foo')
+                    expect(xml.data['basic-csv'].at(1).Col2.text()).to.equal('Bar')
+                });
+            });
+        });
+
+        describe('When the user converts a CSV file which contain a space in the column name into a XML file', () => {
+            beforeEach(() => {
+                return runCommand('node app.js --output=xml --files specs/data/csv-to-xml/csv-spaces-in-columns.csv');
+            });
+
+            afterEach(() => {
+                return deleteFileIfExists('./specs/data/csv-to-xml/csv-spaces-in-columns.xml');
+            });
+
+            it('Should output the XML file as expected', () => {
+                return getXml('./specs/data/csv-to-xml/csv-spaces-in-columns.xml').then(xml => {
+                    expect(xml.data['csv-spaces-in-columns'].count()).to.equal(2);
+                    expect(xml.data['csv-spaces-in-columns'].at(0).Col1.text()).to.equal('Hi')
+                    expect(xml.data['csv-spaces-in-columns'].at(0).Col2.text()).to.equal('Bye')
+                    expect(xml.data['csv-spaces-in-columns'].at(1).Col1.text()).to.equal('Foo')
+                    expect(xml.data['csv-spaces-in-columns'].at(1).Col2.text()).to.equal('Bar')
+                });
+            });
+
+
+        });
+
+        describe('When the user converts a CSV file which contain a different characters in the column name -which need to be escaped- into a XML file', function () {
+            beforeEach(() => {
+                return runCommand('node app.js --output=xml --files specs/data/csv-to-xml/csv-characters-need-be-escaped.csv');
```

Comment:
Maybe rename CSV to be explicit that the invalid characters are in the headers not the data